packages/proxy/src/util/responseHelpers.ts

```diff
@@ -1,27 +1,29 @@
 import type { ServerResponse } from 'node:http';
+import { Readable } from 'node:stream';
 import { pipeline } from 'node:stream/promises';
-import { DiscordAPIError, HTTPError, RateLimitError } from '@discordjs/rest';
-import type { Dispatcher } from 'undici';
+import { DiscordAPIError, HTTPError, RateLimitError, type ResponseLike } from '@discordjs/rest';
 
 /**
  * Populates a server response with the data from a Discord 2xx REST response
  *
  * @param res - The server response to populate
  * @param data - The data to populate the response with
  */
-export async function populateSuccessfulResponse(res: ServerResponse, data: Dispatcher.ResponseData): Promise<void> {
-	res.statusCode = data.statusCode;
+export async function populateSuccessfulResponse(res: ServerResponse, data: ResponseLike): Promise<void> {
+	res.statusCode = data.status;
 
-	for (const header of Object.keys(data.headers)) {
+	for (const [header, value] of data.headers) {
 		// Strip ratelimit headers
-		if (header.startsWith('x-ratelimit')) {
+		if (header.toLowerCase().startsWith('x-ratelimit')) {
```

Comment:
We should use `/^x-ratelimit/i` instead, it's significantly faster than lowercasing the entire string and then checking if it starts with something.